src/main/java/com/jcabi/log/Logger.java

```diff
@@ -429,6 +429,32 @@ public void write(final int data) throws IOException {
     }
 
     /**
+     * Log messages constructed from Suppliers (Java 8 users).
```

Comment:
@mkordas because I don't think it makes sense for other users? They can save the method execution by themselves with an ``if()` and it's less code. The alternative, as you said is passing an implementation of ``Supplier`` - and it would have to be declared on the spot (anonymous class), since there would be one for every logged method call. 
For every logged message you would see a monster like this
```
Logger.withSupplier().debug(this, "Some %s", new Supplier<T>(this){
     T get() {
            this.calculate();
     }
});
```

and it would have the exact behaviour as the normal debug in an if()